File: Exercicio_17/biblioteca/src/entities/Biblioteca.java
package entities;

import java.util.ArrayList;
import java.util.Iterator;
import java.util.List;

public class Biblioteca {
    private List<Livro> livros;

    public Biblioteca() {
        livros = new ArrayList<>();
    }

    public void adicionarLivro(Livro livro) {
        livros.add(livro);
    }

    public void removerLivro(String titulo) {
        Iterator<Livro> iterator = livros.iterator();
        while (iterator.hasNext()) {
            Livro livro = iterator.next();
            if (livro.titulo.equals(titulo)) {
                iterator.remove();
                return;
            }
        }
        System.out.println("Livro não encontrado.");
    }

    public Livro buscarLivro(String titulo) {
        for (Livro livro : livros) {
            if (livro.titulo.equals(titulo)) {
                return livro;
            }
        }
        return null;
    }

    public void listarLivrosDisponiveis() {
        for (Livro livro : livros) {
            if (livro.disponivel) {
                livro.exibirInfo();
                System.out.println();
            }
        }
    }

    public void exibirInformacoes(String titulo) {
        Livro livro = buscarLivro(titulo);
        if (livro != null) {
            livro.exibirInfo();
        } else {
            System.out.println("Livro não encontrado.");
        }
    }
}

File: 17_OrientacaoAObjetos/Biblioteca.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace _17_OrientacaoAObjetos
{
    internal class Biblioteca
    {
        public List<Livro> livros = new List<Livro>();

        public void AdicionarLivro(Livro livro)
        {
            livros.Add(livro);
            Console.WriteLine("Livro adicionado à biblioteca.\n");
        }

        public void RemoverLivro(string titulo)
        {
            Livro livro = BuscarLivro(titulo);

            if (livro != null)
            {
                livros.Remove(livro);
                Console.WriteLine("Livro removido.\n");
            }
            else
            {
                Console.WriteLine("Livro não encontrado.\n");
            }
        }

        public Livro BuscarLivro(string titulo)
        {
            foreach (Livro livro in livros)
            {
                if (livro.Titulo.ToLower() == titulo.ToLower())
                {
                    return livro;
                }
            }

            return null;
        }

        public void ListarLivroDisponiveis()
        {
            Console.WriteLine("\nLivros disponíveis: \n");

            foreach (Livro livro in livros)
            {
                if (livro.Disponivel)
                {
                    Console.WriteLine("- " + livro.Titulo);
                }
            }
            
            Console.WriteLine();

        }

        public void ExibirInformacoes(string titulo)
        {
            Livro livro = BuscarLivro(titulo);

            if (livro != null)
            {
                livro.ExibirInfo();
            }
            else 
            {
                Console.WriteLine("Livro não encontrado.");
            }
        }
    }
}
